plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/AggregationQueryPageSource.java

```diff
@@ -0,0 +1,225 @@
+        }
+        return new Page(blocks);
+    }
+
+    // TODO consider support 6.1/6.2
+    //  6.1/6.2: no after_key, just use last bucket as the next retrieve parameter
+    //  >= 6.3: the result contains after_key, use after_key as the next retrieve
+    private Optional<Map<String, Object>> extractAfter(SearchResponse searchResponse)
+    {
+        Aggregations aggregations = searchResponse.getAggregations();
+        if (aggregations == null) {
+            return Optional.empty();
+        }
+        for (Aggregation agg : aggregations) {
+            if (agg instanceof CompositeAggregation) {
+                CompositeAggregation compositeAggregation = (CompositeAggregation) agg;
+                return Optional.ofNullable(compositeAggregation.afterKey());
+            }
+        }
+        return Optional.empty();
+    }
+
+    private List<Map<String, Object>> getResult(SearchResponse searchResponse)
+    {
+        Aggregations aggregations = searchResponse.getAggregations();
+        if (aggregations == null) {
+            return Collections.emptyList();
+        }
+        Map<String, Object> singleValueMap = new LinkedHashMap<>();
+        ImmutableList.Builder<Map<String, Object>> result = ImmutableList.builder();
+        boolean hasBucketsAggregation = false;
+        boolean hasMetricAggregation = false;
+        for (Aggregation agg : aggregations) {
+            if (agg instanceof CompositeAggregation) {
+                hasBucketsAggregation = true;
+                if (hasMetricAggregation) {
+                    throw new IllegalStateException("Bucket and metric aggregations should not be both present.");
+                }
+                CompositeAggregation compositeAggregation = (CompositeAggregation) agg;
+                for (CompositeAggregation.Bucket bucket : compositeAggregation.getBuckets()) {
+                    Map<String, Object> line = new HashMap<>();
+                    line.putAll(bucket.getKey());
+                    for (Aggregation metricAgg : bucket.getAggregations()) {
+                        if (metricAgg instanceof NumericMetricsAggregation.SingleValue) {
+                            NumericMetricsAggregation.SingleValue sv = (NumericMetricsAggregation.SingleValue) metricAgg;
+                            line.put(metricAgg.getName(), extractSingleValue(sv));
+                        }
+                    }
+                    result.add(line);
+                }
+            }
+            else if (agg instanceof NumericMetricsAggregation.SingleValue) {
+                hasMetricAggregation = true;
+                if (hasBucketsAggregation) {
+                    throw new IllegalStateException("Bucket and metric aggregations should not be both present.");
+                }
+                NumericMetricsAggregation.SingleValue sv = (NumericMetricsAggregation.SingleValue) agg;
+                singleValueMap.put(agg.getName(), extractSingleValue(sv));
+            }
```

Comment:
Since we only pushdown composite aggregation or metric aggregation, so `else` branch must not be happened here.
But considering the completeness, add the `else` branch and throw exception is also good idea.